pkg/storage/secret/metadata/secure_value_store.go

```diff
@@ -199,21 +224,33 @@ func (s *secureValueMetadataStorage) Update(ctx context.Context, newSecureValue
 }
 
 func (s *secureValueMetadataStorage) Delete(ctx context.Context, namespace xkube.Namespace, name string) error {
-
 	// Delete from the keeper.
 	// TODO: here temporary, the moment of deletion will change in the async flow.
 	// TODO: do we care to inform the caller if there is any error?
 	_ = s.deleteFromKeeper(ctx, namespace, name)
 
 	// TODO: do we need to delete by GUID? name+namespace is a unique index. It would avoid doing a fetch.
-	row := &secureValueDB{Name: name, Namespace: namespace.String()}
+	req := deleteSecureValue{
+		SQLTemplate: sqltemplate.New(s.dialect),
+		Namespace:   namespace.String(),
+		Name:        name,
+	}
 
-	err := s.db.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+	query, err := sqltemplate.Execute(sqlSecureValueDelete, req)
+	if err != nil {
+		return fmt.Errorf("execute template %q: %w", sqlSecureValueDelete.Name(), err)
+	}
+
+	err = s.db.GetSqlxSession().WithTransaction(ctx, func(sess *session.SessionTx) error {
```

Comment:
Doesn't need a transaction right?

Reply:
you are right, it is not needed, I removed it